pdm/pep517/base.py

```diff
@@ -130,68 +128,63 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
     def build(self, build_dir: str, **kwargs) -> str:
         raise NotImplementedError
 
-    def _find_files_iter(self, for_sdist: bool = False) -> Iterator[str]:
+    def _get_include_and_exclude_paths(
+        self, for_sdist: bool = False
+    ) -> Tuple[Iterator[str], Iterator[str]]:
         includes = set()
-        find_froms = set()
         excludes = set()
-        dont_find_froms = set()
-        source_includes = self.meta.source_includes or ["tests"]
-        meta_includes = itertools.chain(self.meta.includes, source_includes)
-        meta_excludes = list(self.meta.excludes)
 
+        meta_excludes = list(self.meta.excludes)
+        source_includes = self.meta.source_includes or ["tests"]
         if not for_sdist:
             # exclude source-includes for non-sdist builds
             meta_excludes.extend(source_includes)
 
-        for pat in meta_includes:
-            if os.path.basename(pat) == "*":
-                pat = pat[:-2]
-            if "*" in pat or os.path.isfile(pat):
-                includes.add(pat)
-            else:
-                find_froms.add(pat)
         if not self.meta.includes:
             top_packages = _find_top_packages(self.meta.package_dir or ".")
             if top_packages:
-                find_froms.update(top_packages)
+                includes |= set(top_packages)
```

Comment:
```suggestion
                includes.update(top_packages)
```
This method is preferred.